### ISSUE TITLE
Question: date functions manipulation

### ISSUE THREAD

**Issue by nerro** [hashicorp/terraform#20416]
hey-hey,

I'm trying to create a formatted date in terraform template (1. day of next month). Is it possible? I can find `timestamp` and `timeadd` functions only.

e.g. today is 21. Feb 2019 and I want to create 1 Mar 2019. Thanks for pointing me in the right direction.

**Comment by apparentlymart** [comment]
Hi @nerro,

The standard way to represent time in Terraform configurations is the RFC 3339 date and time syntax, and so any resource type that accepts timestamps as an argument should accept that format; if you've found one that uses a different format then it'd be worth opening an issue with that provider to fix it.

With that said, we know that sometimes you need to format dates for consumption by systems that _aren't_ accessed via Terraform providers, or for humans, and so for the forthcoming v0.12.0 release we've added a [`formatdate`](https://github.com/hashicorp/terraform/blob/0ddbe5d7be7760b8e3a1debce84d947181f5d96d/website/docs/configuration/functions/formatdate.html.md) function that allows custom date formatting, primarily focused on machine-oriented standard timestamp formats, but usable for human-oriented output too if your readers speak English. That functionality is not yet available in any stable release, but will be included in the forthcoming v0.12.0 release which will see its first beta release in the near future.

```
> formatdate("D MMM YYYY", "2018-01-02T23:12:01Z")
2 Jan 2018
```

Until that is released, unfortunately there is no direct feature for this and so it would require some complicated string transformations, e.g. using the `replace` function and a lookup map for month names, or similar.

**Comment by apparentlymart** [comment]
Because this feature is already implemented in the master branch, and because we use GitHub only to track work yet to be completed, I'm going to close this now. Look out for news on the 0.12.0 release where you'll be able to use this new feature.
